src/App.test.js

```diff
@@ -13,32 +20,28 @@ const weekdays = [
   "Saturday",
 ];
 
-function getPostDay({ createdAt }) {
-  return new Date(createdAt).getDay();
-}
-
-// sort posts by weekday (Sunday to Saturday)
-mockPosts.sort((a, b) => getPostDay(a) - getPostDay(b));
-
 test.each(weekdays)(
   "shows table containing correct posts for %s",
   async (weekday) => {
-    const { getByText, getByRole, getAllByRole } = render(<App />);
+    const { getByText, getByRole, getByLabelText, findAllByRole } = render(
+      <App />
+    );
 
-    const weekdayButton = getByText(weekday);
-    fireEvent.click(weekdayButton);
-
-    const day = weekdays.indexOf(weekday);
-    const postIndex = mockPosts.findIndex((post) => getPostDay(post) === day);
+    const select = getByLabelText(/Selected weekday/);
+    fireEvent.change(select, { target: { value: weekday } });
 
+    await waitForElementToBeRemoved(() => getByText(/Loading/));
     getByRole("table");
-    const rows = getAllByRole("row");
 
-    for (let i = 0; i < rows.length; i += 1) {
-      const post = mockPosts[postIndex + i];
-      getByText(post.author);
-      getByText(post.title);
-      getByText(post.score.toString());
-    }
+    const rows = await findAllByRole("row");
+    const day = weekdays.indexOf(weekday);
+    const postsForWeekday = mockPosts.filter((post) => post.day === day);
```

Comment:
Looks like there is no `day` property in the post object, so `postsForWeekday` will be empty. And that is why the tests are passing, because the `forEach` below will never loop :-)